`static_assert` should not be overriden in c++

```
/opt/nxdk/lib/pdclib/platform/xbox/include/assert.h:32:9: warning: keyword is hidden by macro definition [-Wkeyword-macro]
#define static_assert _Static_assert
        ^
```

--- a/platform/xbox/include/assert.h
+++ b/platform/xbox/include/assert.h
@@ -29,7 +29,9 @@ void _xbox_assert(char const * const expression, char const * const file_name, c
                           __LINE__), 0)))
 #endif
 
+#ifndef __cplusplus
 #define static_assert _Static_assert
+#endif
 
 #ifdef __cplusplus
 }

